Tests/UnitTests/Networking/ETagManagerTests.swift

```diff
@@ -86,6 +86,30 @@ class ETagManagerTests: TestCase {
         expect(response?.requestDate) == requestDate
     }
 
+    func testStoredResponseIsNotUsedIfResponseCodeIs304ButContainsNoETag() throws {
+        let eTag = "an_etag"
+        let url = self.urlForTest()
+        let request = URLRequest(url: url)
+        _ = self.mockStoredETagResponse(for: url, statusCode: .success, eTag: eTag)
+        let responseObject = try JSONSerialization.data(withJSONObject: ["a": "response"])
+        let requestDate = Date().addingTimeInterval(-100000)
+
+        let response = self.eTagManager.httpResultFromCacheOrBackend(
+            with: self.responseForTest(url: url,
+                                       body: responseObject,
+                                       eTag: nil,
+                                       statusCode: .notModified,
+                                       requestDate: requestDate),
+            request: request,
+            retried: false
+        )
+
+        expect(response).toNot(beNil())
+        expect(response?.statusCode) == .notModified
+        expect(response?.body) == responseObject
```

Comment:
the response _is_ used in this case, though, right? the name of the test says it's not used, I'm not entirely sure what that means in this context

Reply:
This is the response from the backend, not the stored one (the `ETagManager.httpResultFromCacheOrBackend` still confuses me too).

Reply:
Hmm I wonder, `304` responses usually won't have a body right? I know this test is for a hypothetical situation, but I wonder if the correct behavior here would be to just error the request... If we try to fallback to a hypothetical body in a 304 response it would make it much harder to catch this unexpected situation.

Reply:
Oooh I see what you're saying. Yeah in that case we'd have to fail cause we can't load the etag.
Let me update the test.

Reply:
Okay so our system is handling this well 👍🏻  I'll leave the updated test to cover this scenario though.
Here's what happens:
- First response: 304, but no etag
- Couldn't find stored response
- `HTTPClient` sees this, and retries the request, with `retried` = `true`
- Because it's retried, `ETagManager` doesn't use the etag, so subsequent request can't return 304